src/Servers/Kestrel/test/InMemory.FunctionalTests/Http3/Http3ConnectionTests.cs

```diff
@@ -219,45 +220,69 @@ public async Task StreamPool_MultipleStreamsInSequence_PooledStreamReused()
 
             await Http3Api.InitializeConnectionAsync(_echoApplication);
 
-            var requestStream = await Http3Api.CreateRequestStream();
-            var streamContext1 = requestStream.StreamContext;
+            var streamContext1 = await MakeRequestAsync(0, headers);
+            var streamContext2 = await MakeRequestAsync(1, headers);
 
-            await requestStream.SendHeadersAsync(headers);
-            await requestStream.SendDataAsync(Encoding.ASCII.GetBytes("Hello world 1"), endStream: true);
+            Assert.Same(streamContext1, streamContext2);
+        }
 
-            Assert.False(requestStream.Disposed);
+        [Theory]
+        [InlineData(10)]
+        [InlineData(100)]
+        [InlineData(1000)]
```

Comment:
Are these numbers significant for any reason? Why does testing two requests as above not give us sufficient coverage? I know these are really fast, but how long does it take to complete 1000 requests in sequence?

Reply:
No significance. 1000 takes less than a second. I wanted a test that stresses pooling and makes thread-safety issues fail reliable.